aiida/manage/manager.py

```diff
@@ -136,8 +136,8 @@ def _load_backend(self, schema_check: bool = True, repository_check: bool = True
         # yet known, we issue a warning in the case the repo and database are incompatible. In the future this might
         # then become an exception once we have verified that it is working reliably.
         if repository_check and not profile.is_test_profile:
-            if profile.container_is_initialised:
-                repository_uuid_config = profile.get_repository_container().container_id
+            if profile.is_repository_initialised:
```

Comment:
Is this check still necessary in case `get_repository_uuid()` returns `None` for non-initialised repositories?

Reply:
If not, perhaps we should make this a hidden method, or even remove it entirely? I feel we may be overdoing it with the number of repository methods/properties. 😅

Reply:
Maybe what would be better is that, now `Profile.get_repository` returns a `Repository` instance, we implement those methods there. That should be the correct way of doing things.

Reply:
Indeed, having `Repository.is_initialised` and `Repository.initialise` sounds like a good idea. 👍